packages/wallet-sdk/src/relay/WalletSDKRelay.ts

```diff
@@ -1211,25 +1216,19 @@ export class WalletSDKRelay extends WalletSDKRelayAbstract {
         });
 
         const _cancel = (error?: Error | number) => {
-          if (typeof error === "number") {
```

Comment:
this block of code wasn't optimal.
i tried refactoring it by deduping some paths, but there might be further improvements that can be made to enhance it.